p22/src/figures.rs

```diff
@@ -0,0 +1,181 @@
+#[derive(Default, Debug, Clone, Copy)]
+pub struct Point {
+    x: f64,
+    y: f64,
+}
+
+impl Point {
+    pub fn new(x: f64, y: f64) -> Self {
+        Self { x, y }
+    }
+}
+
+impl Eq for Point {
+    fn assert_receiver_is_total_eq(&self) {}
+}
+
+impl PartialEq for Point {
```

Comment:
You should not implement these traits manually here, instead you should derive them.